slang-gfx.h

```diff
@@ -871,16 +1028,12 @@ class IQueryPool : public ISlangUnknown
     { 0xc2cc3784, 0x12da, 0x480a, { 0xa8, 0x74, 0x8b, 0x31, 0x96, 0x1c, 0xa4, 0x36 } }
 
 
-class ICommandEncoder : public ISlangUnknown
+class ICommandEncoder
```

Comment:
By removing ISlangUnknown reference counting is removed (which could be seen as a benefit) - but you also lose being able to queryInterface, which breaks the main mechanisms for future non breaking binary compatibility. 

Moreover in this specific scenario since it has interfaces deriving from it, it means there is no mechanism to upcast.

Reply:
Having this interface being a ISlangUnknown encourages the use of ComPtr on the user side, which can lead to wrong code since the object is transient and could be deleted when the user calls release via ComPtr. If we ever need queryInterface we should introduce a ITemporary or something like that in the future.